server/channels/api4/cloud.go

```diff
@@ -399,7 +399,7 @@ func getCloudProducts(c *Context, w http.ResponseWriter, r *http.Request) {
 	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadBilling) {
 		sanitizedProducts := []model.UserFacingProduct{}
 		err = json.Unmarshal(byteProductsData, &sanitizedProducts)
-		if err != nil {
+		if err != nil || sanitizedProducts == nil {
```

Comment:
Same. This is unncessary as well. It will just write `null` as the output which is valid if the input is null.